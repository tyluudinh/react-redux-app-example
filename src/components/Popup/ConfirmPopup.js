import React, { Component } from 'react';

import Popup from 'app/components/Popup/Popup';
import Button from 'app/components/Button/Button';

import './ConfirmPopup.css';

export default class ForgotPasswordPopup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ''
    }
    this.show = this.show.bind(this);
    this.hide = this.hide.bind(this);
  }
  
  show() {
    this.popup.show();
  }

  hide() {
    this.popup.hide();
  }

  render() {
    const { title, confirm, confirmText, cancelText, message, negative } = this.props;
    return (
      <Popup 
        style={{ padding: '20px' }}
        innerClass="confirm-popup"
        title={ title }
        ref={(ref) => this.popup = ref}>
        <div>
          <p>{ message }</p>
          <div className="btns-wrapper">
            <Button className="delete-account-pop__button" type={negative ? 'red-reverse' : 'green'} onClick={confirm} minPadding>{ confirmText || 'Confirm' }</Button>
            <Button className="delete-account-pop__button" type="green-reverse" onClick={this.hide} minPadding>{ cancelText || 'Cancel' }</Button>
          </div>
        </div>
      </Popup>
    )
  }
}