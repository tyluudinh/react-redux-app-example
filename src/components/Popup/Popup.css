.popup__wrapper {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.6);
  visibility: hidden;
  opacity: 0;
  transition: 0.3s;
  z-index: 11;
  display: flex;
  flex-direction: column;
  justify-content: center;
  &.active {
    opacity: 1;
    visibility: visible;
  }
}

.popup__inner {
  position: relative;
  width: 100%;
  height: 100%;
  background: white;
  padding: 12px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  p {
    font-size: 0.9rem;
    text-align: center;
  }
  label {
    font-size: 0.9rem;
    display: flex;
    flex-direction: row;
    justify-content: center;
    line-height: 50px;
    .inline-field {
      margin-left: 12px;
    }
  }
  @media screen and (min-width: 33.75em) {
    max-width: 480px;
    height: auto;
    margin: auto;
  }
}

.popup__inner--big {
  max-width: 100%;
  @media screen and (min-width: 33.75em) {
    max-width: 85%;
  }  
}

.popup__close-button {
  position: absolute;
  top: 14px;
  right: 14px;
  fill: #9b9b9b;
  width: 28px;
  height: 28px;
  cursor: pointer;
}

.popup__content {
  overflow: auto;
  .ReactTable {
    overflow: auto;
  }
}

.popup__title {
  font-weight: lighter;
  font-size: 1.5rem;
  text-align: center;
  color: #009247; 
  margin: 6px;
}