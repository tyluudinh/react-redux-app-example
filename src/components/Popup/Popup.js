import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CloseIcon from 'react-icons/lib/md/close';

import './Popup.css';

export default class Popup extends Component {
  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.element,
    innerClass: PropTypes.string,
    bigOne: PropTypes.bool,
  }

  constructor(props) {
    super(props);
    this.state = {
      show: false
    }
    this.show = this.show.bind(this);
    this.hide = this.hide.bind(this);
  }
  show() {
    this.setState({
      show: true
    })
  }
  hide() {
    this.setState({
      show: false
    })
  }
  stopPropagating(e) {
    e.stopPropagation();
  }
  render() {
    const { children, title, innerClass, bigOne } = this.props;
    return (
      <div 
        className={
          classNames(
            'popup__wrapper',
            { 'active': this.state.show }
          )
        }
        onClick={this.hide}
      > 
        <div 
          className={classNames(
            'popup__inner',
            { 'popup__inner--big': bigOne },
            innerClass
          )} 
          onClick={this.stopPropagating}
        >
          <CloseIcon className="popup__close-button" onClick={this.hide} />
          {(title) ? (<h3 className="popup__title">{title}</h3>) : null}
          <div className="popup__content">
            {children}
          </div>
        </div>
      </div>
    )
  }
}