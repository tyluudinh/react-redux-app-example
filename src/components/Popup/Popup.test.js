import React from 'react';

import Popup from './Popup';

describe('Comopnent Popup', () => {
  const BasicPopup = mount(<Popup title="Basic Popup" innerClass="custom-class">Something inside</Popup>);
  const BasicPopupNode = BasicPopup.instance();

  test('Render correctly', () => {
    const BasicPopupJson = toJson(BasicPopup);
    expect(BasicPopupJson).toMatchSnapshot();
  })

  test('Set active class when it\'s showing', () => {
    BasicPopup.setState({ show: true });
    expect(BasicPopup.find('div').at(0).hasClass('active')).toBeTruthy();
  })

  test('Set active class when it\'s showing', () => {
    BasicPopup.setState({ show: false });
    expect(BasicPopup.find('div').at(0).hasClass('active')).toBeFalsy();
  })

  test('show method set the correct state', () => {
    BasicPopupNode.show();
    expect(BasicPopup.state('show')).toBeTruthy();
  })

  test('hide method set the correct state', () => {
    BasicPopupNode.hide();
    expect(BasicPopup.state('show')).toBeFalsy();
  })

  test('Click on inner box does not propagate to outer popup (cause it close)', () => {
    BasicPopupNode.show();
    BasicPopup.find('.popup__inner').simulate('click');
    expect(BasicPopup.state('show')).toBeTruthy();
  });
})