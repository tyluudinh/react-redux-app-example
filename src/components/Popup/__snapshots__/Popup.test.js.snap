// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comopnent Popup Render correctly 1`] = `
<Popup
  innerClass="custom-class"
  title="Basic Popup"
>
  <div
    className="popup__wrapper"
    onClick={[Function]}
  >
    <div
      className="popup__inner custom-class"
      onClick={[Function]}
    >
      <MdClose
        className="popup__close-button"
        onClick={[Function]}
      >
        <IconBase
          className="popup__close-button"
          onClick={[Function]}
          viewBox="0 0 40 40"
        >
          <svg
            className="popup__close-button"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
              />
            </g>
          </svg>
        </IconBase>
      </MdClose>
      <h3
        className="popup__title"
      >
        Basic Popup
      </h3>
      <div
        className="popup__content"
      >
        Something inside
      </div>
    </div>
  </div>
</Popup>
`;
